napari/layers/base/base.py

```diff
@@ -458,10 +461,17 @@ def _mode_setter_helper(self, mode: Union[Mode, str]) -> Mode:
             New mode for the current layer.
 
         """
-        mode = self._modeclass(mode)
+        mode = self._modeclass(mode_in)
+        # Sub-classes can have their own Mode enum, so need to get members
+        # from the specific mode class set on this layer.
+        PAN_ZOOM = getattr(self._modeclass, "PAN_ZOOM", None)
+        TRANSFORM = getattr(self._modeclass, "TRANSFORM", None)
+        assert PAN_ZOOM is not None
+        assert TRANSFORM is not None
```

Comment:
```suggestion
        PAN_ZOOM = getattr(self._modeclass, "PAN_ZOOM")
        TRANSFORM = getattr(self._modeclass, "TRANSFORM")
```
Why not this? Based on python [documentation](https://docs.python.org/3/library/functions.html#getattr) this approach will return a more readable error.

Reply:
I'm not sure, and your way is definitely better 😄